Namespace internal API

TheOneric opened the pull request:
See #222 and #654.
One unsued function is dropped, ~~11~~ 12 are static'fied and possibly moved, 3 are moved into headers as `static inline` and all others renamed. It also adds a check to our CI to avoid adding new non-namespaced internal functions.

There are a few more functions which technically could be moved and static'fied like in the thrid commit, but I either wasn't sure if or don't think it's a good idea. They (might) logically fit better into their current files. See below:
```
--- Not sure if moving would be good ---

apply_transition_effects (parse -> render)
event_has_hard_overrides (parse -> render)

fix_outline (bitmap -> render)
be_padding (bitmap -> render)


--- Probably a bad idea to move ---

resolve_base_direction (shaper -> render)

outline_add_rect (outline -> font)
outline_convert (outline -> font)
outline_rotate_90 (outline -> font)

outline_scale_pow2 (outline -> render)
outline_stroke (outline -> render)
outline_to_bitmap (outline -> render)
outline_transform_2d (outline -> render)
outline_transform_3d (outline -> render)
outline_update_cbox (outline -> render)
outline_update_min_transformed_x (outline -> render)

rasterizer_done (rasterizer -> render)
rasterizer_init (rasterizer -> render)
rasterizer_set_outline (rasterizer -> bitmap)

copy_bitmap (bitmap -> render)
realloc_bitmap (bitmap -> blur)
shift_bitmap (bitmap -> render)
```

MrSmile commented:
There's also a third method: make function `static inline` and put it into appropriate header. I think `skip_spaces`, `rskip_spaces`, `change_alpha`, `be_padding` and maybe `numpad2align` should be handled in that way.

Also, `be_blur_pre` and `be_blur_post` are good candidates for assembly implementation, so maybe it's better to use assembly naming scheme for them instead, similarly to `ass_be_blur`.

TheOneric commented:
Good idea re `static inline` in headers! I converted `skip_spaces`, `rskip_spaces` and  `numpad2align` to those. `be_padding` is only used in `ass_render.c`, so if its moved out of `ass_bitmap.c` anyway, I just moved it directly to `ass_render.c` instead of a header.

`change_alpha` though calls another function `static` to `ass_parse.c` which also calls another `static` function. To convert `change_alpha` to header `static inline` those would need to be moved as well, eventhough they're not used anywhere else which imho doesn't seem worth it.
I always get confused about non-static inline with its differences in GNU89 and ISO-C99/GNU99, but it seems like with `change_alpha` being explicitly `extern` in the header and the definition in `ass_parse.c` marked as `inline`, the intnent and likely effect is that hthe compiler gets a hint to and can inline usage in `ass_parse.c`, but an external version is also compiled which is then used in other compilation units. Given `change_alpha` is only called once outside of `ass_parse.c` there's probably also not much perf to be gained from turning it into `static inline`.

> Also, `be_blur_pre` and `be_blur_post` are good candidates for assembly implementation, so maybe it's better to use assembly naming scheme for them instead, similarly to `ass_be_blur`.

Do you mean the `ass_` prefix, the `_c` suffix or both? Imho it's fine as is for now. The suffix only makes sense if there's already an alternative version. For them to be used in `BitmapEngine`, I believe the functions will need to be un`static`fied at which point the prefix can be added — and the CI check will enforce this.

MrSmile commented:
`ass_change_alpha` looks ugly as it's no longer complimentary to `change_color`, also `inline` without `static` is questionable. Maybe it's better to make
```c
void ass_apply_fade(uint32_t *clr, int fade)
{
    // VSFilter compatibility: apply fade only when it's positive
    if (fade > 0)
        change_alpha(clr, mult_alpha(_a(*clr), fade), 1);
}
```
and export it instead? There's no need to move `mult_alpha` in that case too.

Also there's a lot of misalignment, where indent of second and later lines in prototypes and calls is no longer corresponds to first line after inserting `ass_` prefix.

MrSmile commented on libass/ass_bitmap.c:
```diff
@@ -122,7 +122,7 @@ void ass_synth_blur(const BitmapEngine *engine, Bitmap *bm,
 }
 
-bool alloc_bitmap(const BitmapEngine *engine, Bitmap *bm,
+bool ass_alloc_bitmap(const BitmapEngine *engine, Bitmap *bm,
```
Missing indent.

MrSmile commented on libass/ass_bitmap.c:
```diff
@@ -154,29 +154,29 @@ void ass_free_bitmap(Bitmap *bm)
 }
 
-bool outline_to_bitmap(ASS_Renderer *render_priv, Bitmap *bm,
+bool ass_outline_to_bitmap(ASS_Renderer *render_priv, Bitmap *bm,
```
Indent.

MrSmile commented on libass/ass_bitmap.c:
```diff
@@ -202,12 +202,12 @@ bool outline_to_bitmap(ASS_Renderer *render_priv, Bitmap *bm,
     bm->top  = y_min;
 
-    if (!rasterizer_fill(render_priv->engine, rst, bm->buffer,
+    if (!ass_rasterizer_fill(render_priv->engine, rst, bm->buffer,
```
Indent.

MrSmile commented on libass/ass_outline.c:
```diff
@@ -245,7 +245,7 @@ bool outline_convert(ASS_Outline *outline, const FT_Outline *source)
  * and coordinates should be in the allowable range.
  */
-void outline_add_rect(ASS_Outline *outline,
+void ass_outline_add_rect(ASS_Outline *outline,
```
Indent.

MrSmile commented on libass/ass_outline.c:
```diff
@@ -345,11 +345,11 @@ bool outline_rotate_90(ASS_Outline *outline, ASS_Vector offs)
  * Source outline can be NULL.
  */
-bool outline_scale_pow2(ASS_Outline *outline, const ASS_Outline *source,
+bool ass_outline_scale_pow2(ASS_Outline *outline, const ASS_Outline *source,
```
Indent.

MrSmile commented on libass/ass_outline.c:
```diff
@@ -397,15 +397,15 @@ bool outline_scale_pow2(ASS_Outline *outline, const ASS_Outline *source,
  * Source outline can be NULL.
  */
-bool outline_transform_2d(ASS_Outline *outline, const ASS_Outline *source,
+bool ass_outline_transform_2d(ASS_Outline *outline, const ASS_Outline *source,
```
Indent.

MrSmile commented on libass/ass_outline.c:
```diff
@@ -432,15 +432,15 @@ bool outline_transform_2d(ASS_Outline *outline, const ASS_Outline *source,
  * Source outline can be NULL.
  */
-bool outline_transform_3d(ASS_Outline *outline, const ASS_Outline *source,
+bool ass_outline_transform_3d(ASS_Outline *outline, const ASS_Outline *source,
```
Indent.

MrSmile commented on libass/ass_outline.c:
```diff
@@ -469,7 +469,7 @@ bool outline_transform_3d(ASS_Outline *outline, const ASS_Outline *source,
  * \brief Find minimal X-coordinate of control points after perspective transform
  */
-void outline_update_min_transformed_x(const ASS_Outline *outline,
+void ass_outline_update_min_transformed_x(const ASS_Outline *outline,
```
Indent.

MrSmile commented on libass/ass_outline.c:
```diff
@@ -1485,11 +1485,11 @@ static bool close_contour(StrokerState *str, int dir)
  * \return false on allocation failure
  */
-bool outline_stroke(ASS_Outline *result, ASS_Outline *result1,
+bool ass_outline_stroke(ASS_Outline *result, ASS_Outline *result1,
```
Indent.

MrSmile commented on libass/ass_outline.h:
```diff
@@ -87,40 +87,40 @@ typedef struct {
-                                      const double m[3][3],
-                                      int32_t *min_x);
-void outline_update_cbox(const ASS_Outline *outline, ASS_Rect *cbox);
+void ass_outline_update_min_transformed_x(const ASS_Outline *outline,
```
One space is missing.

MrSmile commented on libass/ass_parse.c:
```diff
@@ -280,7 +280,7 @@ static ASS_Style *lookup_style_strict(ASS_Track *track, char *name, size_t len)
  * \param pwr multiplier for some tag effects (comes from \t tags)
  */
-char *parse_tags(ASS_Renderer *render_priv, char *p, char *end, double pwr,
+char *ass_parse_tags(ASS_Renderer *render_priv, char *p, char *end, double pwr,
```
Indent.

MrSmile commented on libass/ass_rasterizer.c:
```diff
@@ -262,7 +262,7 @@ static bool add_cubic(RasterizerData *rst, const ASS_Vector *pt)
 
 
-bool rasterizer_set_outline(RasterizerData *rst,
+bool ass_rasterizer_set_outline(RasterizerData *rst,
```
Indent.

MrSmile commented on libass/ass_rasterizer.c:
```diff
@@ -719,7 +719,7 @@ static bool rasterizer_fill_level(const BitmapEngine *engine, RasterizerData *rs
 }
 
-bool rasterizer_fill(const BitmapEngine *engine, RasterizerData *rst,
+bool ass_rasterizer_fill(const BitmapEngine *engine, RasterizerData *rst,
```
Indent.

MrSmile commented on libass/ass_render.c:
```diff
@@ -1239,26 +1239,26 @@ size_t ass_outline_construct(void *key, void *value, void *priv)
 
             ASS_Outline src;
-            if (!outline_scale_pow2(&src, &k->outline->outline[0],
+            if (!ass_outline_scale_pow2(&src, &k->outline->outline[0],
```
Indent.

MrSmile commented on libass/ass_render.c:
```diff
@@ -1239,26 +1239,26 @@ size_t ass_outline_construct(void *key, void *value, void *priv)
                                     k->scale_ord_x, k->scale_ord_y))
                 return 1;
-            if (!outline_stroke(&v->outline[0], &v->outline[1], &src,
+            if (!ass_outline_stroke(&v->outline[0], &v->outline[1], &src,
```
Indent.

MrSmile commented on libass/ass_render.c:
```diff
@@ -2650,12 +2650,12 @@ size_t ass_composite_construct(void *key, void *value, void *priv)
         v->bm.left += last->pos.x;
         v->bm.top  += last->pos.y;
-    } else if (n_bm && alloc_bitmap(render_priv->engine, &v->bm,
+    } else if (n_bm && ass_alloc_bitmap(render_priv->engine, &v->bm,
```
Indent.

MrSmile commented on libass/ass_render.c:
```diff
@@ -2677,10 +2677,10 @@ size_t ass_composite_construct(void *key, void *value, void *priv)
         v->bm_o.left += last_o->pos_o.x;
         v->bm_o.top  += last_o->pos_o.y;
-    } else if (n_bm_o && alloc_bitmap(render_priv->engine, &v->bm_o,
+    } else if (n_bm_o && ass_alloc_bitmap(render_priv->engine, &v->bm_o,
```
Indent.

MrSmile reviewed (approved):
I've marked places with incorrect alignment, otherwise LGTM.

TheOneric commented:
All indents should now be adjusted.
